### ISSUE TITLE
Train

### ISSUE THREAD

**Issue by Mauricio-Molina24** [ultralytics/ultralytics#21812]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I want to use the yolov8n model to train my segmentation model, and then convert it to onnx format so I can run it on a raspberrypi4b, but it is very heavy, so I wanted to know if there is any way to make this model lighter, since I tried to do a training, doing transferlearning, freezing only the first 10 layers of the model, and then another training that no longer uses the pre-trained yolov8n-seg.pt model but the last trained model from the previous training, but it did not improve much.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Mauricio-Molina24, thanks for opening this issue and for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
-Your exact training/export commands and full logs (including warnings)
-Model/checkpoint used (e.g., .pt path), target format (e.g., ONNX), and resulting file size
-Target device details (e.g., Raspberry Pi 4B specs) and any performance measurements
-Dataset details with a few example images and labels
-And verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)
These details will help us give focused guidance on making your segmentation model and exported artifact lighter for your deployment 👍

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Training different ways wouldn't change the speed of the model.

You can try exporting to NCNN to get faster speeds.

https://docs.ultralytics.com/guides/raspberry-pi/#use-ncnn-on-raspberry-pi

**Comment by Mauricio-Molina24** [comment]
Yes, it's something I haven't tried, thank you very much :)

**Comment by glenn-jocher** [comment]
Great—when you try NCNN, use the smallest seg model and reduce input size for Pi 4B; for example:
```bash
yolo export model=yolo11n-seg.pt format=ncnn imgsz=480
yolo predict model=yolo11n-seg_ncnn_model source=0
```
See the [NCNN export guide](https://docs.ultralytics.com/integrations/ncnn/) for details and options.
